tests/virt_autotest/virsh_internal_snapshot.pm

```diff
@@ -35,6 +35,13 @@ sub run_test {
     foreach my $guest (keys %xen::guests) {
         my $type = check_guest_disk_type($guest);
         next if ($type == 1);
+        record_info "virsh-snapshot", "Cleaning in case of rerun";
+        if (script_run("virsh snapshot-list $guest | grep internal-snapshot-$guest-01") != 0) {
+            assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-01 || true";
+        }
+        if (script_run("virsh snapshot-list $guest | grep internal-snapshot-$guest-02") != 0) {
```

Comment:
If grep is successful(return code is 0), then the delete snapshot should be executed, right? So it should check "if(script_run(xxx) = 0)"?